bioio/writers/ome_zarr_writer.py

```diff
@@ -1,431 +1,478 @@
+
+        if param is None:
+            if not required:
+                return [None] * levels
+            # repeat the default tuple for every level
+            default = default_fn(self.level_shapes[0])
+            return [default] * levels
+
+        # Normalize into a list of length `levels`
+        items: List[Tuple[int, ...]]
+        if isinstance(param, list):
+            items = param
         else:
-            pixelsizes = (
-                physical_pixel_sizes.Z if physical_pixel_sizes.Z is not None else 1.0,
-                physical_pixel_sizes.Y if physical_pixel_sizes.Y is not None else 1.0,
-                physical_pixel_sizes.X if physical_pixel_sizes.X is not None else 1.0,
-            )
-        if channel_names is None:
-            # TODO this isn't generating a very pretty looking name but it will be
-            # unique
-            channel_names = (
-                [
-                    generate_ome_channel_id(image_id=image_name, channel_id=i)
-                    for i in range(image_data.shape[cdimindex])
-                ]
-                if cdimindex > -1
-                else [generate_ome_channel_id(image_id=image_name, channel_id=0)]
-            )
-        if channel_colors is None:
-            # TODO generate proper colors or confirm that the underlying lib can handle
-            # None
-            channel_colors = (
-                [i for i in range(image_data.shape[cdimindex])]
-                if cdimindex > -1
-                else [0]
-            )
-        # Chunk spatial dimensions
-        scale_dim_map = {
-            biob.dimensions.DimensionNames.Time: 1.0,
-            biob.dimensions.DimensionNames.Channel: 1.0,
-            biob.dimensions.DimensionNames.SpatialZ: pixelsizes[0],
-            biob.dimensions.DimensionNames.SpatialY: pixelsizes[1],
-            biob.dimensions.DimensionNames.SpatialX: pixelsizes[2],
-        }
-        transforms = [
-            [
-                # the voxel size for the first scale level
-                {
-                    "type": "scale",
-                    "scale": [scale_dim_map[d] for d in dimension_order],
-                }
-            ]
+            items = [param] * levels
+
+        if len(items) != levels:
+            raise ValueError(f"Length of {name} list must be {levels}")
+
+        # Clamp each item to the level shape
+        return [
+            tuple(min(items[i][d], self.level_shapes[i][d]) for d in range(5))
+            for i in range(levels)
         ]
-        # TODO precompute sizes for downsampled also.
-        plane_size = (
-            image_data.shape[xdimindex]
-            * image_data.shape[ydimindex]
-            * image_data.itemsize
-        )
 
-        target_chunk_size = 16 * (1024 * 1024)  # 16 MB
-        # this is making an assumption of chunking whole XY planes.
+    @staticmethod
+    def _init_store(store: Union[str, zarr.storage.StoreLike]) -> zarr.Group:
```

Comment:
If no store is provided can we create one for the user using this?

Reply:
yeah this is the same functionality as the V2 writer